core/src/main/java/org/infinispan/metrics/impl/VendorAdditionalMetrics.java

```diff
@@ -23,7 +23,9 @@ public void bindTo(MeterRegistry registry) {
       List<BufferPoolMXBean> bufferPoolBeans = ManagementFactory.getPlatformMXBeans(BufferPoolMXBean.class);
       for (BufferPoolMXBean bufferPoolBean : bufferPoolBeans) {
          String name = bufferPoolBean.getName();
-         Gauge.builder(PREFIX + "BufferPool.used.memory." + name, bufferPoolBean, BufferPoolMXBean::getMemoryUsed)
+         // avoid illegal characters due to beans named for example: "mapped - 'non-volatile memory'"
+         String metricName = NameUtils.filterIllegalChars(name).toLowerCase();
```

Comment:
Maybe we could improve this, moving any trailing underscore. For instance here `mapped - 'non-volatile memory'` is translated into `mapped_non_volatile_memory_` instead of `mapped_non_volatile_memory`

Reply:
Thank you for the feedback, the updated PR now removes leading and trailing underscores.